src/gsi/m_fedNode.F90

```diff
@@ -0,0 +1,265 @@
+module m_fedNode
+!$$$  subprogram documentation block
+!                .      .    .                                       .
+! subprogram:    module m_fedNode
+!   prgmmr:      YPW <yaping.wang@ou.ed>
+!      org:      CIMMS 
+!     date:      2019-09-24
+!
+! abstract: class-module of obs-type fedNode (GLM flash extent density)
+! Modified based on m_tdNode.f90
+!
+! program history log:
+!   2019-09-24  YPW   - added this document block for the initial polymorphic
+!                         implementation.
+!
+!   input argument list: see Fortran 90 style document below
+!
+!   output argument list: see Fortran 90 style document below
+!
+! attributes:
+!   language: Fortran 90 and/or above
+!   machine:
+!
+!$$$  end subprogram documentation block
+
+! module interface:
+  use m_obsdiagNode, only: obs_diag
+  use m_obsdiagNode, only: obs_diags
+  use kinds , only: i_kind,r_kind
+  use mpeu_util, only: assert_,die,perr,warn,tell
+  use m_obsNode, only: obsNode
+
+  implicit none
+  private
+
+  public:: fedNode
+
+  type,extends(obsNode):: fedNode
+     !type(td_ob_type),pointer :: llpoint => NULL()
+     type(obs_diag), pointer :: diags => NULL() 
+     real(r_kind)    :: res           !  flash extent density residual
+     real(r_kind)    :: err2          !  flash extent density error squared
+     real(r_kind)    :: raterr2       !  square of ratio of final obs error 
+                                      !  to original obs error
+     !real(r_kind)    :: time          !  observation time in sec     
+     real(r_kind)    :: b             !  variational quality control parameter
+     real(r_kind)    :: pg            !  variational quality control parameter
+     real(r_kind)    :: jb            !  variational quality control parameter
+     real(r_kind)    :: wij(8)        !  horizontal interpolation weights
+     real(r_kind)    :: fedpertb        !  random number adding to the obs
+!     logical         :: luse          !  flag indicating if ob is used in pen.
+     integer(i_kind) :: k1            !  level of errtable 1-33
+     integer(i_kind) :: kx            !  ob type
+     integer(i_kind) :: ij(8)         !  horizontal locations
+
+!     integer(i_kind) :: idv,iob	      ! device id and obs index for sorting
+     real   (r_kind) :: dlev            ! reference to the vertical grid
+     !real   (r_kind) :: elat, elon      ! earth lat-lon for redistribution
+     !real   (r_kind) :: dlat, dlon      ! earth lat-lon for redistribution
+  contains
+    procedure,nopass::  mytype
+    procedure::  setHop => obsNode_setHop_
+    procedure::   xread => obsNode_xread_
+    procedure::  xwrite => obsNode_xwrite_
+    procedure:: isvalid => obsNode_isvalid_
+    procedure::  gettlddp => gettlddp_
+
+    ! procedure, nopass:: headerRead  => obsHeader_read_
+    ! procedure, nopass:: headerWrite => obsHeader_write_
+    ! procedure:: init  => obsNode_init_
+    ! procedure:: clean => obsNode_clean_
```

Comment:
Line 56, 58, 59, 68-71: remove if not needed

Reply:
Agreed.